capstone/src/main/java/com/google/sps/data/YoutubeCaller.java

```diff
@@ -0,0 +1,35 @@
+
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.io.IOException;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/** This class calls the Youtube Data API */
+public class YoutubeCaller implements ApiCaller {
```

Comment:
This class is identical to `PerspectiveCaller`, so maybe we should combine them into a single class `ApiCaller` and then rename the interface to `PostRequest` or something like that?

Reply:
addressed in "address nits"